openlibrary/plugins/openlibrary/js/index.js

```diff
@@ -20,52 +18,46 @@ import '../../../../vendor/js/flot/jquery.flot.stack.js';
 import '../../../../vendor/js/flot/jquery.flot.pie.js';
 import { validateEmail, validatePassword } from './account.js';
 import autocompleteInit from './autocomplete';
+// Used only by the openlibrary/templates/books/edit/addfield.html template
 import addNewFieldInit from './add_new_field';
 import automaticInit from './automatic';
-import { getAvailabilityV2,
-    updateBookAvailability, updateWorkAvailability } from './availability';
 import bookReaderInit from './bookreader_direct';
 import { ungettext, ugettext,  sprintf } from './i18n';
 import addFadeInFunctionsTojQuery from './jquery.customFade';
 import jQueryRepeat from './jquery.repeat';
-import './jquery.scrollTo';
 import { enumerate, htmlquote, websafe, foreach, join, len, range } from './jsdef';
 import { plot_minigraph, plot_tooltip_graph } from './plot';
 import initAnalytics from './ol.analytics';
-import init, { closePop, bookCovers, isScrolledIntoView } from './ol.js';
+import init  from './ol.js';
 import * as Browser from './Browser';
 import { commify } from './python';
-import { Subject, urlencode, renderTag, slice } from './subjects';
+import { Subject, urlencode, slice } from './subjects';
 import Template from './template.js';
-// Add $.fn.focusNextInputField, $.fn.ol_confirm_dialog, $.fn.tap
-import { closePopup, initShowPasswords, truncate, cond } from './utils';
+// Add $.fn.focusNextInputField, $.fn.ol_confirm_dialog
+import { closePopup, truncate, cond } from './utils';
 import initValidate from './validate';
 import '../../../../static/css/js-all.less';
 // polyfill Promise support for IE11
 import Promise from 'promise-polyfill';
 
 // Eventually we will export all these to a single global ol, but in the mean time
 // we add them to the window object for backwards compatibility.
-window.bookCovers = bookCovers;
-window.closePop = closePop;
+// Can be removed once removed from templates. noop function to avoid JS errors.
+window.bookCovers = () => {};
```

Comment:
@cdrini does this work for you? It will avoid JS errors in the unlikely event that the code is not being used.

Reply:
Wait, why do you says the function doesn't do anything? It looks like it does something :/

Reply:
Did you see my comment on your new PR? I basically want to shove all this dead code used by `advanced_search.html` into `advanced_search.html`, so that we can remove it all once that's done. Putting it here feels unnecessary if we do that.

Reply:
Okay so to be clear you want to keep the bookCovers method as is?

I was just a little confused as I am not sure how you are testing this code so I didn't think it mattered. Are you saying it's definitely running on advanced search page? How can I replicate that locally?

Reply:
To clarify, my main goal is to ensure there are 0 references to something before we remove it. So in this PR, yes, I want to keep `bookCovers` as-is simply because it's referenced in other files. That's how I'm "testing" whether it's safe to delete. Although I can't run `advanced_search.html`, it is being used there since it's directly referenced. Note we've done this in the past: https://github.com/internetarchive/openlibrary/commit/878008a66c82f159ed4cd9efe4e1da3e9cb33b74

Reply:
Okay I've kept this in tact as I think that's what you wanted. I will add the removal to #2397 (which will get a rebase conflict when this is merged that I will deal with)

Reply:
^^ That's a valid argument against a solution by the way; if doing something a certain way adds a ton of time to your work flow, say so; I'm recommending this approach because I don't think it has a high development cost; please correct me if I'm wrong. What conflicts are there? That PR looks like it touches entirely different files.

Reply:
Keeping bookCovers as is - is fine to me. I'm fairly confident so whether bookCovers does anything or is an empty function the result is the same  (:-)) - we can pursue that topic in #2397